reframe/listener.go

```diff
@@ -310,21 +310,13 @@ func (listener *ReframeListener) Provide(ctx context.Context, pr *client.Provide
 	return ch, nil
 }
 
-func (listener *ReframeListener) deleteCidFromDatastoreAndIndexes(ctx context.Context, c cid.Cid) {
-	err := listener.dsWrapper.deleteCidTimestamp(ctx, c)
-	if err != nil {
-		log.Warnw("Error cleaning up timestamp by cid index. Continuing.", "err", err)
-	}
-
-	listener.cidQueue.removeCidNode(c)
-}
-
 // Revise logic here
-func (listener *ReframeListener) removeExpiredCids(ctx context.Context) error {
+func (listener *ReframeListener) removeExpiredCids(ctx context.Context) (bool, error) {
 	lastElem := listener.cidQueue.nodesLl.Back()
 	currentTime := time.Now()
 	chunksToRemove := make(map[string]*cidsChunk)
 	cidsToRemove := make(map[cid.Cid]struct{})
+	removedSomeCids := false
```

Comment:
nit: Since this is only used in the first `for` loop below, its declaration can be moved into the loop.  In go, if you declare a variable inside a loop, it will get initialized on each iteration.